akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ExternalInteractions.scala

```diff
@@ -50,7 +50,7 @@ private[akka] trait JournalInteractions[C, E, S] {
     // https://github.com/akka/akka/issues/29262
     onWriteInitiated(ctx, cmd, repr)
 
-    val write = AtomicWrite(repr) :: Nil
+    val write = AtomicWrite(metadata.fold(repr)(metadata => repr.withMetadata(metadata))) :: Nil
```

Comment:
This is once for every write so maybe do a pattern match instead of allocate a lambda